test/shopify-cli/task/ensure_env_test.rb

```diff
@@ -14,18 +14,176 @@ def setup
+          api_key: 1235,
+          secret: 1234,
+          shop: "store.myshopify.com",
+          scopes: "write_products,write_customers,write_draft_orders",
+          host: "host",
+          extra: {}
+        ).returns(stub(:write))
+        EnsureEnv.call(@context)
+      end
+
+      def test_not_all_required_values_found
+        Resources::EnvFile.expects(:parse_external_env).with.returns({ api_key: "apikey", secret: "secret" })
+        ShopifyCli::PartnersAPI::Organizations.stubs(:fetch_with_app).returns(partners_api_response)
+        CLI::UI::Prompt.expects(:ask).with(@context.message('core.tasks.ensure_env.organization_select')).returns(101)
+        CLI::UI::Prompt.expects(:ask).with(
+          @context.message('core.tasks.ensure_env.development_store_select')
+        ).returns('store.myshopify.com')
+
+        Resources::EnvFile.expects(:new).with(
+          api_key: 1235,
+          secret: 1234,
+          shop: "store.myshopify.com",
+          scopes: "write_products,write_customers,write_draft_orders",
+          extra: {}
+        ).returns(stub(:write))
+        EnsureEnv.call(@context, required: [:api_key, :secret, :shop])
+      end
+
+      def test_regenerate_with_existing_env_file
+        Resources::EnvFile
+          .expects(:parse_external_env)
+          .returns({ api_key: "apikey", secret: "secret", shop: "shop", host: "host" })
+        ShopifyCli::PartnersAPI::Organizations.stubs(:fetch_with_app).returns(partners_api_response)
+        CLI::UI::Prompt.expects(:ask).with(@context.message('core.tasks.ensure_env.organization_select')).returns(101)
+        CLI::UI::Prompt.expects(:ask).with(
+          @context.message('core.tasks.ensure_env.development_store_select')
+        ).returns('store.myshopify.com')
+
+        Resources::EnvFile.expects(:new).with(
+          api_key: 1235,
+          secret: 1234,
+          shop: "store.myshopify.com",
+          scopes: "write_products,write_customers,write_draft_orders",
+          host: "host",
+          extra: {}
+        ).returns(stub(:write))
+        EnsureEnv.call(@context, regenerate: true)
+      end
+
+      def test_regenerate_no_existing_env_file
+        Resources::EnvFile.expects(:parse_external_env).with.returns({})
+        ShopifyCli::PartnersAPI::Organizations.stubs(:fetch_with_app).returns(partners_api_response)
+        CLI::UI::Prompt.expects(:ask).with(@context.message('core.tasks.ensure_env.organization_select')).returns(101)
+        CLI::UI::Prompt.expects(:ask).with(
+          @context.message('core.tasks.ensure_env.development_store_select')
+        ).returns('store.myshopify.com')
+
+        Resources::EnvFile.expects(:new).with(
+          api_key: 1235,
+          secret: 1234,
+          shop: "store.myshopify.com",
+          scopes: "write_products,write_customers,write_draft_orders",
+          extra: {}
+        ).returns(stub(:write))
+        EnsureEnv.call(@context, regenerate: true)
+      end
```

Comment:
I think we are missing tests when all required values exists in the .env file such that `Resources::EnvFile.expects(:new).never`?

Reply:
I see a lot of duplicated code in these test cases.  For example, line 141-144 is repeated many times.  We should extract it into a method to keep things DRY.

Also, I think line 70-76,  127-134, 146-152, and etc can be more DRY as well.  Maybe have a method:
```
def envFileValues() {
   { api_key: 1235, secret: 1234, shop:  "store.myshopify.com", scopes: "..." }
```

When you call it from line 146, it can be something like:
```
Resources::EnvFile.expects(:new).with(envFileValues.merge({extra: {}).returns(...))
```